Topic_3/src/keyPoint1/AmountToTextTest.java

```diff
@@ -0,0 +1,111 @@
+
+	@Test
+	public void testForTens20to99() {
+		CheckAmount checkAmount = new CheckAmount(20f);
+		String textCheck = checkAmount.toText();
+		assertEquals("Twenty and 0/100 dollars", textCheck);
+		checkAmount.set(68);
+		textCheck = checkAmount.toText();
+		assertEquals("Sixty-eight and 0/100 dollars", textCheck);
+		checkAmount.set(71);
+		textCheck = checkAmount.toText();
+		assertEquals("Seventy-one and 0/100 dollars", textCheck);
+	}
+
+	@Test
+	public void testForHundreds() {
+		CheckAmount checkAmount = new CheckAmount(900);
+		String textCheck = checkAmount.toText();
+		assertEquals("Nine hundred and 0/100 dollars", textCheck);
+		checkAmount.set(201);
+		textCheck = checkAmount.toText();
+		assertEquals("Two hundred one and 0/100 dollars", textCheck);
+		checkAmount.set(316);
+		textCheck = checkAmount.toText();
+		assertEquals("Three hundred sixteen and 0/100 dollars", textCheck);
+		checkAmount.set(410);
+		textCheck = checkAmount.toText();
+		assertEquals("Four hundred ten and 0/100 dollars", textCheck);
+		checkAmount.set(666);
+		textCheck = checkAmount.toText();
+		assertEquals("Six hundred sixty-six and 0/100 dollars", textCheck);
+		checkAmount.set(699);
+		textCheck = checkAmount.toText();
+		assertEquals("Six hundred ninety-nine and 0/100 dollars", textCheck);
+	}
+
+	@Test
+	public void testWithCents() {
+		CheckAmount checkAmount = new CheckAmount(10.53);
+		String textCheck = checkAmount.toText();
+		assertEquals("Ten and 53/100 dollars", textCheck);
+		checkAmount.set(201.78);
+		textCheck = checkAmount.toText();
+		assertEquals("Two hundred one and 78/100 dollars", textCheck);
+		checkAmount.set(99.99);
+		textCheck = checkAmount.toText();
+		assertEquals("Ninety-nine and 99/100 dollars", textCheck);
+		checkAmount.set(416.10);
+		textCheck = checkAmount.toText();
+		assertEquals("Four hundred sixteen and 10/100 dollars", textCheck);
+	}
+
+	@Test
+	public void testGeneral(){
+		CheckAmount checkAmount = new CheckAmount(9000.23);
+		String textCheck = checkAmount.toText();
+		assertEquals("Nine thousand and 23/100 dollars", textCheck);
+		checkAmount.set(2010.12);
+		textCheck = checkAmount.toText();
+		assertEquals("Two thousand ten and 12/100 dollars", textCheck);
+		checkAmount.set(3016.44);
+		textCheck = checkAmount.toText();
+		assertEquals("Three thousand sixteen and 44/100 dollars", textCheck);
+		checkAmount.set(4100);
+		textCheck = checkAmount.toText();
+		assertEquals("Four thousand one hundred and 0/100 dollars", textCheck);
+		checkAmount.set(958123.33);
+		textCheck = checkAmount.toText();
+		assertEquals("Nine hundred fifty-eight thousand one hundred twenty-three and 33/100 dollars", textCheck);
+		checkAmount.set(10845100.13);
+		textCheck = checkAmount.toText();
+		assertEquals("Ten million eight hundred forty-five thousand one hundred and 13/100 dollars", textCheck);
+		checkAmount.set(1000000.13);
+		textCheck = checkAmount.toText();
+		assertEquals("One million and 13/100 dollars", textCheck);
+	}

```

Comment:
Don’t make unnecessary assertions 
Which specific behaviour are you testing? It’s counterproductive to Assert() anything that’s also asserted by another test: it just increases the frequency of pointless failures without improving unit test coverage one bit. This also applies to unnecessary Verify() calls – if it isn’t the core behaviour under test, then stop making observations about it! Sometimes, TDD folks express this by saying “have only one logical assertion per test”. 
Remember, unit tests are a design specification of how a certain behaviour should work, not a list of observations of everything the code happens to do.